nautobot/extras/models/models.py

```diff
@@ -196,7 +197,24 @@ def get_config_context(self):
             config_context_data = ConfigContext.objects.get_for_object(self).values_list("data", flat=True)
         else:
             config_context_data = self.config_context_data or []
+            # Device and VirtualMachine's Location has its own ConfigContext and its parent Locations' ConfigContext, if any, should
+            # also be applied. However, since moving from mptt to django-tree-queries https://github.com/nautobot/nautobot/issues/510,
+            # we lost the ability to query the ancestors for a particular tree node for subquery https://github.com/matthiask/django-tree-queries/issues/54.
+            # So instead of constructing the location related query in ConfigContextModelQueryset._get_config_context_filters(), which is complicated across databases
+            # We append the missing parent location query here as a patch.
+            if self._meta.model_name == "device":
+                location_config_context_queryset = ConfigContext.objects.filter(
+                    Q(locations__in=self.location.ancestors(include_self=True)) | Q(locations=None),
```

Comment:
Why the `| Q(locations=None)`? This is causing config contexts with no location to apply to all devices. I created a config context with tenant=xyz and no location and it is being rendered for a device with a location and no tenant

Reply:
It seems like all the constructed queries from `nautobot/extras/queryset.py` in line 41 have this `Q(locations=None)`. So I thought it was necessary to include it.

Reply:
I'm actually seeing all contexts rendered even if the location is completely different. Here's my location hierarchy:

- Campus-01
  - Building-02
    - Floor-03
      - Room-05
        - Aisle-13
        - Aisle-06

My device is in Aisle-13 but it's rendering a context applied to Aisle-06

![image](https://github.com/nautobot/nautobot/assets/75227981/b6b8278f-946c-4352-a58f-cd37bd5403f5)

![image](https://github.com/nautobot/nautobot/assets/75227981/e858b64e-0027-41bd-b9ba-54ee7be8b4d2)

Reply:
The latest commit should fix that issue if you want to test it out on your end.